### ISSUE TITLE
'unload' handler is not called

### ISSUE THREAD

**Issue by jjagoda** [rikschennink/conditioner#91]
I'm trying to subscribe to `'unload'` event of modules initialized on selected DOM nodes.
My implementation is based on example shown here: http://conditionerjs.com/docs/#conditioner-sync
```javascript
    // Find period element on the page
    var periodElement = document.querySelector('.peroid');

    // Initialize all datepicker modules
    // within the period element
    var datePickerNodes = conditioner.parse(periodElement);

    // Synchronize load events, we only want to work
    // with these modules if they are all loaded
    var syncGroup = conditioner.sync(datePickerNodes);

    // Wait for load event to fire
    Observer.subscribe(syncGroup,'load',function(nodes){

        // All modules now loaded

    });

    // Also listen for unload event
    Observer.subscribe(syncGroup,'unload',function(nodes){

        // One of the modules has unloaded

    });
```

The effects are:
* when modules load the `'load'` event if fired and the handler is called - this works fine
* when modules unload the handler for `'unload'` does not fire - this isn't working as it should

The call stack is briefly as follows: 

1. 'unload' event is fired - `ModuleController`'s `_unload` publishes `'unload'` event
1. `Observer.publish` goes to line that executes matched callbacks (`matches[i].fn(data);`)
1. goes into one of the callbacs which is `NodeController._onModuleUnload` (this empties the `receivers` array from `Observer.publish`)
1. goes back to `Observer.publish`
1. `receivers` are now empty so it doesn't go into the for loop (which would publish `'unload'`)

If I remove the line with `Observer.conceal(moduleController,this);` from `NodeController`'s `_onModuleUnload` handler the `'unload'` handler is working fine.

If you compare `_onModuleUnload` with `_onModuleLoad` then you'll see that `Observe.conceal` isn't called in the onload handler. What's the purpose of calling it in `_onModuleUnload`?

**Comment by rikschennink** [comment]
Thanks for the detailed report, I'm going to take a look at it as soon as possible.

**Comment by rikschennink** [comment]
I've updated the tests (they failed to reproduce this problem in the first place) and fixed the _onModuleUnload method. This branch should solve the issue:
https://github.com/rikschennink/conditioner/tree/fix/syncgroup-unload

**Comment by jjagoda** [comment]
Rik, thanks for the fix.
These are my first days with conditioner so I couldn't help more then describing what actually happens.

I'll check the fix soon.

**Comment by rikschennink** [comment]
You're welcome :)

**Comment by rikschennink** [comment]
Merged in version 1.2.2
https://github.com/rikschennink/conditioner/tree/1.2.2
